[wasm] [sdb] Initial PR towards making some of sdb's code shareable.

kumpera opened the pull request:
This PR adds the new files, moves the enums / structs and a small bit of code to validate the change.

vargaz commented on mono/mini/debugger-agent.c:
```diff
@@ -286,79 +287,6 @@ typedef enum {
-	EVENT_KIND_KEEPALIVE = 14,
-	EVENT_KIND_USER_BREAK = 15,
-	EVENT_KIND_USER_LOG = 16
-} EventKind;
```
Does this have to move too, doesn't it belong to the debugger protocol ?

kumpera replied:
The problem is that those 3 enums are used deep in the breakpoint/single step code.

Let me try to see how much would need to be shuffled around to move them back.

vargaz commented on mono/mini/debugger-agent.c:
```diff
@@ -286,79 +287,6 @@ typedef enum {
-	SUSPEND_POLICY_EVENT_THREAD = 1,
-	SUSPEND_POLICY_ALL = 2
-} SuspendPolicy;
-
```
Same.

vargaz commented on mono/mini/debugger-agent.c:
```diff
@@ -286,79 +287,6 @@ typedef enum {
-	ERR_INVOKE_ABORTED = 107,
-	ERR_LOADER_ERROR = 200, /*XXX extend the protocol to pass this information down the pipe */
-} ErrorCode;
-
```
Same.

lewurm commented:
is `mono/mini/mini-wasm-debugger.c` supposed to be an empty file?

kumpera commented:
@lewurm yes, this PR just add makefile scaffold and move a bit of code around to validate the change.

kumpera commented:
@vargaz I moved SuspendPolicy and ErrorCode back to sdb.
EventKind will be quite some work as it's stored in EventRequest and that gets shared across.
I added a FIXME to later deal with it properly. I think that's an ok compromise for now.

kumpera commented:
Last commit fixes the wasm breakage.

kumpera commented:
@monojenkins build Linux i386
